master/pkg/model/experiment.go

```diff
@@ -463,3 +463,74 @@ type SearcherEvent struct {
 	EventType    string  `db:"event_type"`
 	Content      JSONObj `db:"content"`
 }
+
+// MetricType denotes what type of step (training / validation) a metric is from.
+type MetricType int
+
+const (
+	// TrainingMetric designates metrics from training steps.
+	TrainingMetric MetricType = iota
+	// ValidationMetric designates metrics from validation steps.
+	ValidationMetric MetricType = iota
+)
+
+// HPImportanceTrialData is the input to the hyperparameter importance algorithm.
+type HPImportanceTrialData struct {
+	TrialID int                    `db:"trial_id"`
+	Hparams map[string]interface{} `db:"hparams"`
+	Batches int                    `db:"batches"`
+	Metric  float64                `db:"metric"`
+}
+
+// ExperimentHPImportance is hyperparameter importance for an experiment, and consists of
+// independent measurements of importance for any of the metrics recorded by the experiment.
+type ExperimentHPImportance struct {
+	TrainingMetrics   map[string]MetricHPImportance `json:"training_metrics"`
+	ValidationMetrics map[string]MetricHPImportance `json:"validation_metrics"`
+}
+
+// MetricHPImportance is hyperparameter importance with respect to a specific metric.
+type MetricHPImportance struct {
+	Error              string             `json:"error"`
+	Pending            bool               `json:"pending"`
+	InProgress         bool               `json:"in_progress"`
+	ExperimentProgress float64            `json:"experiment_progress"`
+	HpImportance       map[string]float64 `json:"hp_importance"`
+}
+
+// SetMetricHPImportance is a convenience function when modifying results for a specific metric.
+func (hpi *ExperimentHPImportance) SetMetricHPImportance(metricHpi MetricHPImportance,
+	metricName string, metricType MetricType) *ExperimentHPImportance {
+	switch metricType {
+	case TrainingMetric:
+		hpi.TrainingMetrics[metricName] = metricHpi
+	case ValidationMetric:
+		hpi.ValidationMetrics[metricName] = metricHpi
+	default:
+		panic("Invalid metric type!")
+	}
+	return hpi
+}
+
+// GetMetricHPImportance is a convenience function when working with results for a specific metric.
+func (hpi *ExperimentHPImportance) GetMetricHPImportance(metricName string, metricType MetricType,
+) MetricHPImportance {
+	switch metricType {
+	case TrainingMetric:
+		metricHpi, ok := hpi.TrainingMetrics[metricName]
+		if !ok {
+			var newMetricHpi MetricHPImportance
+			metricHpi = newMetricHpi
+		}
+		return metricHpi
```

Comment:
maybe reads a little less confusing
```suggestion
if metricHpi, ok := hpi.TrainingMetrics[metricName]; ok {
	return metricHpi
} else {
	return MetricHPImportance{}
}
```